packages/fetch-router/src/lib/file-handler.test.ts

```diff
@@ -0,0 +1,1206 @@
+import * as assert from 'node:assert/strict'
+import { describe, it, mock } from 'node:test'
+import SuperHeaders, { type SuperHeadersInit } from '@remix-run/headers'
+
+import { createFileHandler } from './file-handler.ts'
+import type { RequestContext } from './request-context.ts'
+import type { RequestMethod } from './request-methods.ts'
+import { AppStorage } from './app-storage.ts'
+
+describe('createFileHandler', () => {
+  function createMockFile(
+    content: string,
+    options: {
+      type?: string
+      lastModified?: number
+    } = {},
+  ): File {
+    return new File([content], 'mock.txt', {
+      type: options.type || 'text/plain',
+      lastModified: options.lastModified || Date.now(),
+    })
+  }
+
+  function createContext(
+    url: string,
+    options: {
+      method?: RequestMethod
+      headers?: SuperHeadersInit
+    } = {},
+  ): RequestContext<'GET', {}> {
+    let headers = new SuperHeaders(options.headers ?? {})
+    return {
+      formData: undefined,
+      storage: new AppStorage(),
+      url: new URL(url),
+      files: null,
+      method: 'GET',
+      request: new Request(url, {
```

Comment:
I think you can just `return new RequestContext(request)` here instead of manually mocking out all properties of a request context object.